vncviewer/ServerDialog.cxx

```diff
@@ -123,6 +140,14 @@ void ServerDialog::run(const char* servername, char *newservername)
 
   strncpy(newservername, dialog.serverName->value(), VNCSERVERNAMELEN);
   newservername[VNCSERVERNAMELEN - 1] = '\0';
+
+  vector<string>::iterator elem = std::find(dialog.serverHistory.begin(), dialog.serverHistory.end(), newservername);
+  // avoid duplicates in the history
+  if(dialog.serverHistory.end() == elem) {
+    dialog.serverHistory.push_back(newservername);
+    dialog.saveServerHistory();
+  }
```

Comment:
Shouldn't we limit the maximum number of entries? I assume the interface gets a bit annoying after a while otherwise. And there might be privacy issues.